libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -4483,13 +4488,54 @@ instance BaseUniverse fn => Functions (ListFn fn) fn where
             let args = appendList (mapList (\(Value a) -> Lit a) pre) (v' :> mapList (\(Value a) -> Lit a) suf)
              in Let (App (injectFn fn) args) (x :-> p)
     FoldMap f | NilCtx HOLE <- ctx -> typeSpec (ListSpec Nothing [] TrueSpec TrueSpec $ FoldSpec f spec)
+    SingletonList | NilCtx HOLE <- ctx -> case spec of
+      TrueSpec -> TrueSpec
+      MemberSpec xss -> MemberSpec [a | [a] <- xss]
+      TypeSpec (ListSpec _ m sz e f) cant
+        -- TODO: fix these error messages!
+        | length m > 1 -> ErrorSpec $ pure "singleton 1"
+        | not $ 1 `conformsToSpec` sz -> ErrorSpec $ pure "singleton 2"
+        | FoldSpec{} <- f -> ErrorSpec $ pure "singleton 3" -- TODO: this is just lazyness for now
+        | any (not . (`conformsToSpec` e)) m -> ErrorSpec $ pure "singleton 4"
```

Comment:
Do we not maintain ``all (`conformsToSpec` elemSpec) must``?

Reply:
Not in the soundness tests...